src/util/taproot.rs

```diff
@@ -90,6 +90,12 @@ impl TapTweakHash {
         }
         TapTweakHash::from_engine(eng)
     }
+
+    /// Converts a `TapTweakHash` into a `Scalar` ready for use with key tweaking API.
+    pub fn to_scalar(&self) -> Scalar {
```

Comment:
Perhaps motivating to add something similar to `ThirtyTwoByteHash` but for scalar. (`trait ToScalar { fn to_scalar(&self) -> Scalar; } impl<T: ThirtyTwoByteHash> ToScalar for T { /* ... */ }`) and accept `T: ToScalar` in tweaking fns. But maybe it's too many generics?

Reply:
I'm going to leave this for now, mainly because I have a weeks worth of notifications to work through.